pulp_smash/pulp_smash_cli.py

```diff
@@ -0,0 +1,177 @@
+# coding=utf-8
+"""The entry point for Pulp Smash's command line interface."""
+import json
+
+import click
+
+from pulp_smash import config, exceptions
+from pulp_smash.config import PulpSmashConfig
+
+
+def _raise_settings_not_found():
+    """Raise `click.ClickException` for settings file not found."""
+    result = click.ClickException(
+        'there is no settings file. Use `pulp-smash settings create` to '
+        'create one.'
+    )
+    result.exit_code = -1
+    raise result
+
+
+@click.group()
+def pulp_smash():
+    """Pulp Smash facilitates functional testing of Pulp."""
+
+
+@pulp_smash.group()
+@click.pass_context
+def settings(ctx):
+    """Manage settings file."""
+    cfg = PulpSmashConfig()
+    try:
+        cfg_path = cfg.get_config_file_path()
+    except exceptions.ConfigFileNotFoundError:
+        cfg_path = None
+    ctx.obj = {
+        'cfg_path': cfg_path,
+        'default_cfg_path': cfg.default_config_file_path,
+    }
+
+
+@settings.command('create')
+@click.pass_context
+def settings_create(ctx):
+    """Create a settings file."""
+    path = ctx.obj['cfg_path']
+    if path:
+        click.echo('Settings file already exist, continuing will override it.')
+        click.confirm('Do you want to continue?', abort=True)
+    else:
+        path = ctx.obj['default_cfg_path']
+    pulp_username = click.prompt('Pulp admin username', default='admin')
+    pulp_password = click.prompt('Pulp admin password', default='admin')
+    pulp_version = click.prompt('Pulp version')
+    system_hostname = click.prompt('System hostname')
+    if click.confirm('Is Pulp published using HTTP?'):
```

Comment:
I think this question is a misleading. This setting determines how Pulp Smash talks to Pulp's API. Whether repositories are published over HTTP or HTTPS is determined when creating a distributor. Maybe this would be better?

```python
system_api_scheme = click.prompt('Pulp API scheme', default='https')
```